src/iosrv/init.c

```diff
@@ -225,17 +234,83 @@ dss_tgt_nr_get(int ncores, int nr)
 static int
 dss_topo_init()
 {
+	uint depth;
+	int num_obj;
+	int num_cores_visited;
+	char *cpuset;
+	int k;
+	hwloc_obj_t corenode;
+
 	hwloc_topology_init(&dss_topo);
 	hwloc_topology_load(dss_topo);
 
 	dss_core_depth = hwloc_get_type_depth(dss_topo, HWLOC_OBJ_CORE);
 	dss_core_nr = hwloc_get_nbobjs_by_type(dss_topo, HWLOC_OBJ_CORE);
-	dss_tgt_nr = dss_tgt_nr_get(dss_core_nr, nr_threads);
+	depth = hwloc_get_type_depth(dss_topo, HWLOC_OBJ_NUMANODE);
+	num_obj = hwloc_get_nbobjs_by_depth(dss_topo, depth);
```

Comment:
minor, numa_node_nr probably is a better name.

Reply:
Changed.